tests/cupyx_tests/distributed_tests/comm_runner.py

```diff
@@ -46,91 +46,118 @@ def run_broadcast(rank, n_workers, root, dtype):
         comm.broadcast(in_array, root)
         testing.assert_allclose(in_array, expected)
 
-    _launch_workers(N_WORKERS, run_broadcast, (0, dtype))
-    _launch_workers(N_WORKERS, run_broadcast, (1, dtype))
+    if use_mpi:
+        from mpi4py import MPI
+        # This process was run with mpiexec
+        run_broadcast(MPI.COMM_WORLD.Get_rank(), 0, dtype, False)
+        run_broadcast(MPI.COMM_WORLD.Get_rank(), 1, dtype, False)
+    else:
+        _launch_workers(run_broadcast, (0, dtype))
+        _launch_workers(run_broadcast, (1, dtype))
```

Comment:
Can't we do this inside `_launch_workers` to avoid code dup/branching? Something like
```
def _launch_workers(func, args, ..., use_mpi=False):
    if use_mpi:
        from mpi4py import MPI
        func(...)
    else:
        # the old _launch_workers impl
```